pkg/metricscollector/v1beta1/file-metricscollector/file-metricscollector_test.go

```diff
@@ -19,40 +19,60 @@ package sidecarmetricscollector
 import (
 	"os"
 	"path/filepath"
-	"reflect"
 	"testing"
 	"time"
 
+	"github.com/google/go-cmp/cmp"
 	commonv1beta1 "github.com/kubeflow/katib/pkg/apis/controller/common/v1beta1"
 	v1beta1 "github.com/kubeflow/katib/pkg/apis/manager/v1beta1"
 	"github.com/kubeflow/katib/pkg/controller.v1beta1/consts"
 )
 
-var testJsonDataPath = filepath.Join("testdata", "JSON")
+const (
+	invalidFileName   = "invalid"
+	invalidFileFormat = "invalid"
 
-func TestCollectObservationLog(t *testing.T) {
+	validJSONTestFile            = "good.json"
+	invalidFormatJSONTestFile    = "invalid-format.json"
+	invalidTimestampJSONTestFile = "invalid-timestamp.json"
+	missingMetricJSONTestFile    = "missing-objective-metric.json"
+
+	validTEXTTestFile            = "good.log"
+	invalidValueTEXTTestFile     = "invalid-value.log"
+	invalidFormatTEXTTestFile    = "invalid-format.log"
+	invalidTimestampTEXTTestFile = "invalid-timestamp.log"
+	missingMetricTEXTTestFile    = "missing-objective-metric.log"
+)
+
+var (
+	testJsonDataPath = filepath.Join("testdata", "JSON")
+	testTextDataPath = filepath.Join("testdata", "TEXT")
+)
 
-	if err := generateTestFiles(); err != nil {
+func TestCollectObservationLog(t *testing.T) {
+	if err := generateTestDirs(); err != nil {
+		t.Fatal(err)
+	}
+	if err := generateJSONTestFiles(); err != nil {
+		t.Fatal(err)
+	}
+	if err := generateTEXTTestFiles(); err != nil {
```

Comment:
Did you push the commit? I can still see 2 separate functions.

Reply:
Yes, I pushed the commit. I just integrated dir creation & deletion operations into functions, test file generation not included.